datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -0,0 +1,133 @@
+
+@Service
+public class SdxHorizontalScalingService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxHorizontalScalingService.class);
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
+
+    @Inject
+    private SdxReactorFlowManager sdxReactorFlowManager;
+
+    @Inject
+    private StackV4Endpoint stackV4Endpoint;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    @Inject
+    private EnvironmentClientService environmentClientService;
+
+    @Inject
+    private RegionAwareInternalCrnGeneratorFactory regionAwareInternalCrnGeneratorFactory;
+
+    @Inject
+    private WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor;
+
+    public FlowIdentifier horizontalScaleDatalake(String name, DatalakeHorizontalScaleRequest scaleRequest) {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        if (!entitlementService.isDatalakeHorizontalScaleEnabled(accountId)) {
+            throw new BadRequestException("Datalake horizontal scale is not enabled");
+        }
+
+        LOGGER.info("Horizontal scaling Datalake with name {}", name);
+        SdxCluster sdxCluster = sdxClusterRepository.findByAccountIdAndClusterNameAndDeletedIsNullAndDetachedIsFalse(accountId, name)
+                .orElseThrow(() -> notFound("SDX cluster", name).get());
+        return sdxReactorFlowManager.triggerHorizontalScaleDataLake(sdxCluster, scaleRequest);
+    }
+
+    public String triggerScalingFlow(SdxCluster sdxCluster, StackScaleV4Request scaleRequest) {
+        try {
+            return ThreadBasedUserCrnProvider.doAsInternalActor(
+                    regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+                    () -> stackV4Endpoint.putScaling(0L, sdxCluster.getClusterName(), scaleRequest, sdxCluster.getAccountId())
+            ).getPollableId();
+        } catch (javax.ws.rs.NotFoundException e) {
+            LOGGER.info("Cannot find stack on Cloudbreak side {}", sdxCluster.getClusterName());
+            throw new RuntimeException("Cannot find stack on Cloudbreak side.", e);
+        } catch (WebApplicationException e) {
+            String errorMessage = webApplicationExceptionMessageExtractor.getErrorMessage(e);
+            LOGGER.info("Cannot horizontal scale stack {} from cloudbreak: {}", sdxCluster.getStackId(), errorMessage, e);
+            throw new RuntimeException(errorMessage);
+        } catch (ProcessingException e) {
+            throw new RuntimeException(e.getMessage());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void validateHorizontalScaleRequest(SdxCluster sdxCluster, DatalakeHorizontalScaleRequest request) {
+        if (!ENTERPRISE.equals(sdxCluster.getClusterShape())) {
+            throw new BadRequestException(String.format("scaling not supported on: %s.", sdxCluster.getClusterShape().name()));
```

Comment:
+ add phrase that instructs customers that "only Datalake Enterprise supports horizontal scale."